extensions/amp-3d-gltf/0.1/amp-3d-gltf.js

```diff
@@ -0,0 +1,206 @@
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getIframe, preloadBootstrap} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {listenFor, postMessage} from '../../../src/iframe-helper';
+import {removeElement} from '../../../src/dom';
+
+const isWebGLSupported = () => {
+  const canvas = document.createElement('canvas');
+  const gl = canvas.getContext('webgl')
+      || canvas.getContext('experimental-webgl');
+  return gl && gl instanceof WebGLRenderingContext;
+};
+
+export class Amp3dGltf extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {!Deferred} */
+    this.willBeReady_ = new Deferred();
+
+    /** @private {!Deferred} */
+    this.willBeLoaded_ = new Deferred();
+
+    /** @private {!JsonObject} */
+    this.context_ = dict();
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    preloadBootstrap(this.win, this.preconnect);
+    this.preconnect.url('https://cdnjs.cloudflare.com/ajax/libs/three.js/91/three.js', opt_onLayout);
+    this.preconnect.url('https://cdn.jsdelivr.net/npm/three@0.91/examples/js/loaders/GLTFLoader.js', opt_onLayout);
+    this.preconnect.url('https://cdn.jsdelivr.net/npm/three@0.91/examples/js/controls/OrbitControls.js', opt_onLayout);
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+
+    this.willBeReady_ = new Deferred();
+    this.willBeLoaded_ = new Deferred();
+
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+    const getOption = (name, fmt, dflt) =>
+      this.element.hasAttribute(name)
+        ? fmt(this.element.getAttribute(name))
+        : dflt;
+
+    const bool = x => x !== 'false';
+    const string = x => x;
+    const number = x => parseFloat(x);
+
+    this.context_ = dict({
+      'src': getOption('src', string, ''),
+      'renderer': {
+        'alpha': getOption('alpha', bool, false),
+        'antialias': getOption('antialiasing', bool, true),
+      },
+      'maxPixelRatio':
+          getOption('maxPixelRatio', number, devicePixelRatio || 1),
+      'controls': {
+        'enableZoom': getOption('enableZoom', bool, true),
+        'autoRotate': getOption('autoRotate', bool, false),
+      },
+      'hostUrl': this.win.location.href,
+    });
+
+    if (!isWebGLSupported()) {
+      this.toggleFallback(true);
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+
+    const iframe = getIframe(
+        this.win, this.element, '3d-gltf', this.context_
+    );
+
+    this.applyFillContent(iframe, true);
+    this.iframe_ = iframe;
```

Comment:
can your viewer go to fullscreen? if so, please add `iframe.setAttribute('allowfullscreen', 'true');`

Reply:
Currently it cannot.